src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java

```diff
@@ -0,0 +1,279 @@
+        if(Objects.isNull(entity) || !entity.getRequestEndpoint().equals(REQUEST_PAYMENTS_XPAY)){
+            log.error("No xPay request entity has been found for requestId: " + requestId);
+            XPayPollingResponseError error = new XPayPollingResponseError(404L, NOT_FOUND_MSG);
+            return createXPayAuthPollingResponse(HttpStatus.NOT_FOUND, error, null);
+        }
+        return  createXPayAuthPollingResponse(HttpStatus.OK, null, entity);
+    }
+
+    private ResponseEntity<XPayAuthResponse> createxPayAuthResponse(String errorMessage, HttpStatus status, String requestId) {
+        XPayAuthResponse response = new XPayAuthResponse();
+        if(Objects.nonNull(requestId)) {
+            PaymentRequestEntity entity = paymentRequestRepository.findByGuid(requestId);
+            response.setRequestId(requestId);
+            response.setStatus(entity.getStatus());
+        }
+        if(StringUtils.isEmpty(errorMessage)) {
+            String urlRedirect = StringUtils.join(PGS_RESPONSE_URL_REDIRECT, requestId);
+            response.setUrlRedirect(urlRedirect);
+        } else {
+            response.setError(errorMessage);
+            log.info("END POST - " + REQUEST_PAYMENTS_XPAY);
+        }
+        return ResponseEntity.status(status).body(response);
+    }
+
+    private ResponseEntity<XPayAuthResponse> createAuthPaymentXPay(XPayAuthRequest pgsRequest, String clientId, String mdcFields) {
+       log.info("START - XPay Request Payment Authorization ");
+
+        PaymentRequestEntity paymentRequestEntity = new PaymentRequestEntity();
+        String transactionId = pgsRequest.getIdTransaction();
+        try{
+            AuthPaymentXPayRequest xPayRequest = createXPayRequest(pgsRequest);
+            String authRequestJson = OBJECT_MAPPER.writeValueAsString(xPayRequest);
+            generateRequestEntity(clientId, mdcFields, transactionId, paymentRequestEntity, xPayRequest.getTimeStamp());
+            paymentRequestEntity.setJsonRequest(authRequestJson);
+            if (Objects.nonNull(XPAY_RESPONSE_URL)) {
+                xPayRequest.setUrlRisposta(String.format(XPAY_RESPONSE_URL, paymentRequestEntity.getGuid()));
+            }
+            else {
+                log.warn("xPayResponseUrl is null");
+                throw new Exception();
+            }
+
+            executeXPayAuthorizationCall(xPayRequest, paymentRequestEntity, transactionId);
+
+        } catch (JsonProcessingException e) {
+            log.error(SERIALIZATION_ERROR_MSG, e);
+            return  createxPayAuthResponse(GENERIC_ERROR_MSG + transactionId, HttpStatus.INTERNAL_SERVER_ERROR, null);
+        } catch (Exception e ) {
+            log.error(GENERIC_ERROR_MSG + transactionId);
+            return createxPayAuthResponse(GENERIC_ERROR_MSG + transactionId, HttpStatus.INTERNAL_SERVER_ERROR, null);
+        }
+
+        log.info("END - XPay Request Payment Authorization ");
```

Comment:
Always make reference to the transactionId
```suggestion
       log.info("END - XPay Request Payment Authorization for idTransaction " + pgsRequest.getIdTransaction());
```